File: src/HackWars/java/game/computerfunctions/requestFTPUpdate.java
package game.computerfunctions;

import game.*;
import assignments.*;
/**
 A function that is run within Computer.java.

 This function requests that the client side pulls an update of its FTP directories.
 */

import java.util.*;

import assignments.*;
import hackscript.model.*;

public class requestFTPUpdate extends function {
    public requestFTPUpdate(Computer MyComputer) {
        super(MyComputer);
    }

    public void execute(ApplicationData MyApplicationData) {
        PacketAssignment PA = super.getComputer().getPacketAssignment();
        PA.setRequestPrimary(true, 8);
        PA.setRequestSecondary(true, 8);
        super.getComputer().sendPacket();
    }
}

File: src/GameServer/java/com/hackwars/game/functions/RequestFTPUpdate.kt
package com.hackwars.game.functions

import game.ApplicationData
import game.Computer

/**
 * Represents a function that requests a full FTP directory refresh.
 *
 * This function marks both primary and secondary FTP directory requests in the
 * packet assignment and then sends a packet so the client pulls fresh directory data.
 *
 * @constructor Initializes `RequestFTPUpdate` for the provided computer.
 *
 * @param computer The `Computer` instance on which this function operates.
 */
class RequestFTPUpdate(computer: Computer) : Function(computer) {
    override fun execute(applicationData: ApplicationData) {
        val packetAssignment = computer.packetAssignment
        packetAssignment.setRequestPrimary(true, 8)
        packetAssignment.setRequestSecondary(true, 8)
        computer.sendPacket()
    }
}
